### DISCUSSION TITLE
マルチプレイSDKをAndroidで使用した際にExecutionEngineExceptionが発生する

### DISCUSSION THREAD

**Discussion by NON906** [pixiv/vroid-sdk-developers#51] [Q&A]
お世話になっております。

マルチプレイSDKをAndroidで使用したいのですが、サンプルをそのまま利用してもモデルデータの表示ができませんでした。
その際、logcatでログを確認したところ、以下のような例外が発生しておりました。

```
09-11 21:19:41.390  9940  9963 E Unity   : ExecutionEngineException: Attempting to call method 'System.Runtime.Serialization.XmlObjectSerializerWriteContext::GetHasValue<System.DateTime>' for which no ahead of time (AOT) code was generated.
09-11 21:19:41.390  9940  9963 E Unity   :   at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <00000000000000000000000000000000>:0
09-11 21:19:41.390  9940  9963 E Unity   :   at System.Runtime.Serialization.XmlFormatWriterInterpreter.UnwrapNullableObject (System.Func`1[TResult] memberValue, System.Type& memberType, System.Boolean& isNull) [0x00000] in <00000000000000000000000000000000>:0
09-11 21:19:41.390  9940  9963 E Unity   :   at System.Runtime.Serialization.XmlFormatWriterInterpreter.WriteValue (System.Type memberType, System.Object memberValue, System.Boolean writeXsiType) [0x00000] in <00000000000000000000000000000000>:0
09-11 21:19:41.390  9940  9963 E Unity   :   at System.Runtime.Serialization.XmlFormatWriterInterpreter.WriteMembers (System.Runtime.Serialization.ClassDataContract classContract, System.
```

この例外は、ダウンロードの進捗のプログレスバーが完了した直後に発生しました。
また、直前に「update storage」というログがあり、おそらくその処理の最中に発生したものだと思われます。

何か解決方法はありますでしょうか？

使用しているバージョンは以下の通りです。
Unity 2021.3.7f1 および 2020.3.20f1
VRoidSDK-v0.1.2
VRoidSDKMultiplayExtension-v0.1.1

よろしくお願いします。

**Comment by nyamadan** [comment]
回答が遅くなり申し訳ございません。

> また、直前に「update storage」というログがあり、おそらくその処理の最中に発生したものだと思われます。

ご指摘の通り、保存後にストレージの更新に失敗していると予想できます。

> ExecutionEngineException: Attempting to call method 'System.Runtime.Serialization.XmlObjectSerializerWriteContext::GetHasValue<System.DateTime>' for which no ahead of time (AOT) code was generated.

とありますので意図しないリフレクションが実行されてしまっている可能性があります。

VRoid SDKの不具合かと思いますので調査いたします。

**Reply by wcl1207** [reply]
@NON906 @nyamadan 
その後は解決されましたでしょうか。
可能であれば、簡単でも良いですが、解決方法を教えていただけないでしょうか。

**Reply by nyamadan** [reply]
調査に時間がかかってしまい申し訳ございません。

こちら、 #89 と同様の原因かと思います。
[CachedModelData.csの修正](https://github.com/pixiv/vroid-sdk-developers/discussions/89#discussioncomment-6195968)をお試しいただけますでしょうか？
